core/src/main/java/org/apache/druid/math/expr/Expr.java

```diff
@@ -824,7 +888,12 @@ public BindingDetails analyzeInputs()
     }
 
     lambdaBindingDetails = lambdaExpr.analyzeInputs();
-    bindingDetails = accumulator.with(lambdaBindingDetails).withArrayArguments(function.getArrayInputs(argsExpr));
+    boolean isArrayOutput = function.hasArrayOutput() || (function instanceof ApplyFunction.BaseFoldFunction
```

Comment:
Using `instanceof` here seems like a code smell -- is there a nice way to add a method to `ApplyFunction` that gets the necessary info out?

Reply:
Maybe? Will have a look

Reply:
I've modified `ApplyFunction.hasArrayOutput` to now take the `LambaExpr` as an argument so `BaseFoldFunction` can just override `hasArrayOutput` to examine the lambda expression to see if it's output is array typed.